hydra-node/src/Hydra/Chain/Blockfrost/Client.hs

```diff
@@ -321,29 +322,85 @@ toCardanoGenesisParameters bfGenesis =
     , _genesisSecurityParam
     } = bfGenesis
 
-mkEraHistory :: Blockfrost.Genesis -> EraHistory
-mkEraHistory genesis = EraHistory (mkInterpreter summary)
+mkEraHistory :: BlockfrostClientT IO EraHistory
+mkEraHistory = do
+  eras' <- Blockfrost.getNetworkEras
+  let eras = filter withoutEmptyEra eras'
+  let summary = mkEra <$> eras
+  case nonEmptyFromList summary of
+    Nothing ->
+      liftIO $ throwIO $ BlockfrostError "Failed to create EraHistory."
+    Just s -> pure $ EraHistory (mkInterpreter $ Summary s)
  where
-  Blockfrost.Genesis
-    { _genesisNetworkMagic
-    , _genesisSystemStart
-    , _genesisSlotLength
-    , _genesisEpochLength
-    } = genesis
-
-  summary :: Summary (CardanoEras StandardCrypto)
-  summary =
-    Summary . NonEmptyOne $
-      EraSummary
-        { eraStart = initBound
-        , eraEnd = EraUnbounded
-        , eraParams
-        }
-
-  eraParams =
+  mkBound Blockfrost.NetworkEraBound{_boundEpoch, _boundSlot, _boundTime} =
+    Bound
+      { boundTime = RelativeTime _boundTime
+      , boundSlot = SlotNo $ fromIntegral _boundSlot
+      , boundEpoch = EpochNo $ fromIntegral _boundEpoch
+      }
+  mkEraParams Blockfrost.NetworkEraParameters{_parametersEpochLength, _parametersSlotLength, _parametersSafeZone} =
     EraParams
-      { eraEpochSize = EpochSize $ fromIntegral _genesisEpochLength
-      , eraSlotLength = mkSlotLength $ fromIntegral _genesisSlotLength
-      , eraSafeZone = UnsafeIndefiniteSafeZone
-      , eraGenesisWin = GenesisWindow 1
+      { eraEpochSize = EpochSize $ fromIntegral _parametersEpochLength
+      , eraSlotLength = mkSlotLength _parametersSlotLength
+      , eraSafeZone = StandardSafeZone _parametersSafeZone
+      , eraGenesisWin = GenesisWindow _parametersSafeZone
+      }
+  mkEra Blockfrost.NetworkEraSummary{_networkEraStart, _networkEraEnd, _networkEraParameters} =
+    EraSummary
+      { eraStart = mkBound _networkEraStart
+      , eraEnd = EraEnd $ mkBound _networkEraEnd
+      , eraParams = mkEraParams _networkEraParameters
       }
+  withoutEmptyEra
+    Blockfrost.NetworkEraSummary
+      { _networkEraStart = Blockfrost.NetworkEraBound{_boundTime = boundStart}
+      , _networkEraEnd = Blockfrost.NetworkEraBound{_boundTime = boundEnd}
+      } = boundStart == 0 && boundEnd == 0
+
+----------------
+-- Wallet API --
+----------------
+
+-- | Query the Blockfrost API for address UTxO and convert to cardano 'UTxO'.
+queryUTxO :: SigningKey PaymentKey -> NetworkId -> BlockfrostClientT IO UTxO
```

Comment:
Should: not require a signing key, but at least a verification key or even better an address. Then, we would not even need `NetworkId`

```suggestion
queryUTxO :: Address ShelleyAddr -> BlockfrostClientT IO UTxO
```